ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java

```diff
@@ -87,7 +87,8 @@ public void downloadPrebuiltModelConfig(String taskId, MLRegisterModelInput regi
                         .url(modelZipFileUrl)
                         .deployModel(deployModel)
                         .modelNodeIds(modelNodeIds)
-                        .modelGroupId(modelGroupId);
+                        .modelGroupId(modelGroupId)
+                        .functionName(FunctionName.from((String) config.get("model_task_type")));;
```

Comment:
Can't we get Function name from `registerModelInput` like the way we get other inputs from 57-62 lines?

Reply:
The registerModelInput is from the request body json. So we can request customer to provide it. But if we want to keep the request convention to only contain "name, version, model_format", we can only read it from pretrained config.

Reply:
What if `config.get("model_task_type")` is null?

Reply:
I feel like may be we should address [this section](https://github.com/opensearch-project/ml-commons/blob/2.x/common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelInput.java#L97-L100)

currently we default to text embedding, which doesn't seem right. What happens if we start adding different pre-trained models like right now we added splade model.

Reply:
I think currently the logic is ok. The only scenario we don't input function name is when we use pretrained model, but if we fix the bug, the function name will be read from config and it is correct. So it's ok to keep a default value.

Reply:
It's ok based on a assumption that, every pre-trained model will have `config.get("model_task_type")`.  I don't know how true that is.

Reply:
> What if `config.get("model_task_type")` is null?

This should be fine, the FunctionName.valueOf is surrounded by try catch, and all exception case will be throwing `Wrong function name`  error.

Reply:
Until now, every pretrained model has its own and we can require the future contributor to have this. I agree we should address this section. But I don't have a clear solution towards this problem.

Reply:
> I feel like may be we should address [this section](https://github.com/opensearch-project/ml-commons/blob/2.x/common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelInput.java#L97-L100)
> 
> currently we default to text embedding, which doesn't seem right. What happens if we start adding different pre-trained models like right now we added splade model.

Can we make the function name mandatory, when it's null, throw exception instead of setting to a default value?

Reply:
one solution could be: we can add function_name in our [model listing](https://artifacts.opensearch.org/models/ml-models/model_listing/pre_trained_models.json) and then get the function name from there.

Reply:
> > I feel like may be we should address [this section](https://github.com/opensearch-project/ml-commons/blob/2.x/common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelInput.java#L97-L100)
> > currently we default to text embedding, which doesn't seem right. What happens if we start adding different pre-trained models like right now we added splade model.
> 
> Can we make the function name mandatory, when it's null, throw exception instead of setting to a default value?

No. When we register pretrained model,  we don't provide function name so in some scenario it would be null. We need a default value.

Reply:
> one solution could be: we can add function_name in our [model listing](https://artifacts.opensearch.org/models/ml-models/model_listing/pre_trained_models.json) and then get the function name from there.

I think both work. Model listing and pretrained config are both files we maintain, so it's just a option to read from which file in our s3 bucket.

Reply:
I see what you mean now. We get this `model_task_type` from [config.json](https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/all-mpnet-base-v2/1.0.1/torch_script/config.json) like this. Yeah this should work.